csvcubed/csvcubed/readers/configdeserialiser.py

```diff
@@ -0,0 +1,594 @@
+        "measure type",
+        "measure types",
+    ],
+    "observations": [
+        "observation",
+        "observations",
+        "obs",
+        "values",
+        "value",
+        "val",
+        "vals",
+    ],
+    "units": ["unit", "units", "units column", "unit column", "unit type", "unit types"]
+    # 'attribute': ['attribute', 'attributes']  # Not supported for loading cube from convention only.
+}
+
+log = logging.getLogger(__name__)
+start_logging()
+
+
+def _load_resource(resource_path: Path) -> dict:
+    """
+    Load a json schema document from either a File or URI
+    """
+    schema: dict = {}
+
+    if "http" in urlsplit(str(resource_path)).scheme:
+        schema = read_from_uri(resource_path, log)
+
+    else:
+        schema = read_json_from_file(resource_path, log)
+
+    return schema
+
+
+# def _get_code_list(
+#     column_label: str,
+#     maybe_code_list: Optional[Union[bool, str]],
+#     json_parent_dir: Path,
+#     maybe_parent_uri: Optional[str],
+#     column_data: PandasDataTypes,
+#     maybe_property_value_url: Optional[str],
+# ) -> Optional[QbCodeList]:
+#
+#     is_date_time_code_list = (
+#         (
+#             maybe_code_list is None
+#             or (isinstance(maybe_code_list, bool) and maybe_code_list is True)
+#         )
+#         and maybe_parent_uri
+#         == "http://purl.org/linked-data/sdmx/2009/dimension#refPeriod"
+#         and maybe_property_value_url is not None
+#         and maybe_property_value_url.startswith("http://reference.data.gov.uk/id/")
+#     )
+#
+#     if is_date_time_code_list:
+#         column_name_csvw_safe = csvw_column_name_safe(column_label)
+#         columnar_data = pandas_input_to_columnar_str(column_data)
+#         concepts = [
+#             DuplicatedQbConcept(
+#                 existing_concept_uri=uritemplate.expand(
+#                     maybe_property_value_url, {column_name_csvw_safe: c}
+#                 ),
+#                 label=c,
+#             )
+#             for c in sorted(set(columnar_data))
+#         ]
+#         return CompositeQbCodeList(
+#             CatalogMetadata(column_label),
+#             concepts,
+#         )
+#
+#     elif maybe_code_list is not None:
+#         if isinstance(maybe_code_list, str):
+#             return ExistingQbCodeList(maybe_code_list)
+#
+#         #elif isinstance(maybe_code_list, bool) and not maybe_code_list:
+#         elif isinstance(maybe_code_list, bool) and maybe_code_list is False:
+#             return None
+#
+#         else:
+#             raise Exception(f"Unexpected codelist value '{maybe_code_list}'")
+#
+#     return NewQbCodeListInCsvW(
+#         json_parent_dir
+#         / "codelists"
+#         / f"{uri_safe(column_label)}.csv-metadata.json"
+#     )
+
+
+def initialise_from_config(self, config_json: dict) -> bool:
+    """
+    Initialises a Qube from a qube json file.
+
+    param: config_json: A Qube created according to a Qube Config schema
+    """
```

Comment:
This function doesn't seem to have an implementation yet. I know that @MuazzamChaud 's work has been merged into main now so it's probably merging his work into your branch and integrating with his CLI `build` command work.

Reply:
I thought you'd gotten rid of this?